### PR TITLE
Numpy array display-able in Spark df.show()

### PR DESCRIPTION
**Pull request by eddyxu**
Print out a summary of `numpy.array` in a Spark DataFrame (#27)

### CODE REVIEW

File: python/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+@pytest.fixture(scope="session")
+def spark() -> SparkSession:
+    # TODO: load rikai jar from "sbt publishLocal" installed version?
+    return (
+        SparkSession.builder.appName("spark-test")
+        .config(
+            "spark.jars",
+            os.path.join(
```
Comment (changhiskhan) [inline python/tests/conftest.py:29]: nit: consider using pathlib instead: 
```
from pathlib import Path

Path(rikai.__file__).parent() / 'jars' / 'rikai_2.12-0.0.1.jar'
```

File: python/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                os.path.dirname(rikai.__file__), "jars", "rikai_2.12-0.0.1.jar"
```
Comment (changhiskhan) [inline python/tests/conftest.py:30]: is there an easy way to pull the current version number? If it's a hassle, just create a github issue and we can wait til after 0.0.1

File: src/main/scala/org/apache/spark/sql/rikai/NDArray.scala [hunk 3]
```diff
@@ -17,10 +17,46 @@
...
+import org.apache.spark.sql.catalyst.InternalRow
 
+/**
+  * NDArray type. Storing the persisted data from numpy ndarray.
+  *
+  * Note that this class does not support interoperation in scala yet.
+  * It will only show a array summary in "df.show()".
+  * But the data is accessible from Pyspark and Pytorch / Tensorflow readers.
+  *
+  * @todo consider to use arrow Array for interchangable data transfer.
```
Comment (changhiskhan) [inline src/main/scala/org/apache/spark/sql/rikai/NDArray.scala:30]: I don't thnk we should invest more in scala interop for numpy ndarray. Let's wait til we're able to take advantage of more Arrow integrations.

File: src/main/scala/org/apache/spark/sql/rikai/NDArray.scala [hunk 4]
```diff
@@ -17,10 +17,46 @@
...
+      case 2  => "uint16"
+      case 3  => "uint32"
+      case 4  => "uint64"
+      case 5  => "int8"
+      case 6  => "int16"
+      case 7  => "int32"
+      case 8  => "int64"
+      case 9  => "float"
+      case 10 => "double"
+      case 11 => "boolean"
```
Comment (changhiskhan) [inline src/main/scala/org/apache/spark/sql/rikai/NDArray.scala:57]: what if the ndarray isn't any of these types (https://numpy.org/devdocs/user/basics.types.html) or if it's a subtype (e.g., datetime64, timedelta64, etc)?

File: python/tests/conftest.py (hunk 2 above)
Comment (eddyxu) [reply python/tests/conftest.py:30]: we can use a `glob` actually.  Do you think we should wait for `sbt publishLocal` to install a `-SNAPSHOT` version?

File: src/main/scala/org/apache/spark/sql/rikai/NDArray.scala (hunk 3 above)
Comment (eddyxu) [reply src/main/scala/org/apache/spark/sql/rikai/NDArray.scala:30]: agreed. so put this into `todo` :)

File: python/tests/conftest.py (hunk 2 above)
Comment (eddyxu) [reply python/tests/conftest.py:30]: I kind lean to not have this `jars` directory in the future.  Even on a Databricks cluster, the jars will be deployed separately from this `pip install rikai`.

File: src/main/scala/org/apache/spark/sql/rikai/NDArray.scala (hunk 4 above)
Comment (eddyxu) [reply src/main/scala/org/apache/spark/sql/rikai/NDArray.scala:57]: what about make it a `str` value , i.e., `"float64"` for now?  Such values should be dictionary encoded by parquet later.

Comment (changhiskhan) [reply src/main/scala/org/apache/spark/sql/rikai/NDArray.scala:57]: sg

**Comment by eddyxu** [comment]
@changhiskhan ready for another round of review.

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**